### PR TITLE
Feature/add possibility to control the combobox component

### PR DESCRIPTION
**Pull request by source-streamer**
Add possibility to control (get/set selected, get number of items) the ComboBox Component

### CODE REVIEW

**Comment by PaulChana** [comment]
@source-streamer Thanks for the PR - Let me go over it with @joefocusrite and we will get back to you ASAP

File: source/cpp/source/DefaultCommandHandler.cpp [hunk 1]
```diff
@@ -361,16 +362,16 @@ std::variant<juce::Slider *, juce::String> getSlider (const Command & command)
...
 
-    auto * slider = dynamic_cast<juce::Slider *> (component);
-    if (slider == nullptr)
-        return "Component is not a slider: " + componentId;
+    auto * castedComponenet = dynamic_cast<T *> (component);
+    if (castedComponenet == nullptr)
+        return "Component is not a " + juce::String(typeid(T).name()) + ": " + componentId;
 
-    return slider;
+    return castedComponenet;
```
Comment (PaulChana) [inline source/cpp/source/DefaultCommandHandler.cpp:369]: There is a spelling mistake here (`Componenet` should be `Component`).

File: source/cpp/source/DefaultCommandHandler.cpp [hunk 2]
```diff
@@ -361,16 +362,16 @@ std::variant<juce::Slider *, juce::String> getSlider (const Command & command)
     if (component == nullptr)
         return "Component not found: " + componentId;
 
-    auto * slider = dynamic_cast<juce::Slider *> (component);
-    if (slider == nullptr)
-        return "Component is not a slider: " + componentId;
+    auto * castedComponenet = dynamic_cast<T *> (component);
+    if (castedComponenet == nullptr)
+        return "Component is not a " + juce::String(typeid(T).name()) + ": " + componentId;
```
Comment (PaulChana) [inline source/cpp/source/DefaultCommandHandler.cpp:367]: Our code standards require spaces before all brackets, so this needs to be:
`return "Component is not a " + juce::String (typeid (T).name ()) + ": " + componentId;`

**Review by PaulChana — CHANGES_REQUESTED** [review]
@source-streamer Really like this solution. There are a couple of very small changes we would request before merging (a spelling mistake and a tiny format issue). One other thing I would request is could you add an example usage to the example app? Not required for merging but does mean that others can see how to use it
Thanks for the PR!

File: source/cpp/source/DefaultCommandHandler.cpp (hunk 1 above)
Comment (source-streamer) [reply source/cpp/source/DefaultCommandHandler.cpp]: done

File: source/cpp/source/DefaultCommandHandler.cpp (hunk 2 above)
Comment (source-streamer) [reply source/cpp/source/DefaultCommandHandler.cpp]: done

**Review by PaulChana — APPROVED** [review]
Very nice work! Thanks for the contribution!

**Pull request merged**